### PR TITLE
Chore: Make ESLint a devDependency/peerDependency (fixes #523)

### LINKED ISSUE #523: Make eslint a peer dependency
`typescript-eslint-parser` is now pulling in `eslint@4.19.1`, ideally it should be asking for a peer dependency of `"eslint": ">=4.19.1"`. This would allow de-duplication of the eslint dependency in consumers of `typescript-eslint-parser` using a different version of `eslint`.

A current workaround is to pin the `eslint` dependency in a consuming project using a resolution. For example, to force using `eslint@^5.0.0`

```json
"resolutions": {
  "eslint": "^5.0.0",
}
```

### PR DESCRIPTION
**Pull request by platinumazure**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,10 +49,10 @@
     "betarelease": "eslint-prerelease beta"
   },
   "dependencies": {
-    "eslint": "4.19.1",
     "typescript-estree": "2.1.0"
   },
   "peerDependencies": {
+    "eslint": "^4.19.0",
```
Comment (ficristo) [inline package.json]: Is it possible to move to eslint 5?

**Comment by JamesHenry** [comment]
@platinumazure I have fixed a conflict after merging in the visitor keys PR, if you are still happy with it please feel free to merge

**Comment by platinumazure** [comment]
Thank you @JamesHenry, I'll merge as all checks have passed!

**Comment by JamesHenry** [comment]
@platinumazure Thanks! I'll cut a release now

**Pull request merged**